features/FEATURE_BLE/source/Gap.cpp

```diff
@@ -113,32 +118,175 @@ void Gap::processConnectionEvent(
-    const uint8_t *advertisingData,
-    PeerAddressType_t addressType
+    PeerAddressType_t peerAddrType,
+    const ConnectionParams_t *connectionParams,
+    const GapScanningParams *scanParams
 ) {
-    AdvertisementCallbackParams_t params;
+    /* Avoid compiler warnings about unused variables. */
+    (void)peerAddr;
+    (void)peerAddrType;
+    (void)connectionParams;
+    (void)scanParams;
+
+    /* Requesting action from porter(s): override this API if this capability
+       is supported. */
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
 
-    memcpy(params.peerAddr, peerAddr, ADDR_LEN);
-    params.rssi = rssi;
-    params.isScanResponse = isScanResponse;
-    params.type = type;
-    params.advertisingDataLen = advertisingDataLen;
-    params.advertisingData = advertisingData;
-    params.peerAddrType = addressType;
+ble_error_t Gap::connect(
+    const BLEProtocol::AddressBytes_t peerAddr,
+    BLEProtocol::AddressType_t peerAddrType,
+    const ConnectionParams_t *connectionParams,
+    const GapScanningParams *scanParams
+) {
+    /* Avoid compiler warnings about unused variables. */
+    (void)peerAddr;
+    (void)peerAddrType;
+    (void)connectionParams;
+    (void)scanParams;
+
+    /* Requesting action from porter(s): override this API if this capability
+       is supported. */
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
 
-    convert_address_type(
-        addressType,
-        peerAddr,
-        params.addressType
-    );
+ble_error_t Gap::disconnect(
+    Handle_t connectionHandle, DisconnectionReason_t reason
+) {
+    /* avoid compiler warnings about unused variables */
+    (void)connectionHandle;
+    (void)reason;
 
-    onAdvertisementReport.call(&params);
+    /* Requesting action from porter(s): override this API if this capability
+       is supported. */
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t Gap::disconnect(DisconnectionReason_t reason) {
+    /* Avoid compiler warnings about unused variables. */
+    (void)reason;
+
+    /* Requesting action from porter(s): override this API if this capability
+       is supported. */
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t Gap::updateConnectionParams(
+    Handle_t handle,
+    const ConnectionParams_t *params
+) {
+    /* avoid compiler warnings about unused variables */
+    (void)handle;
+    (void)params;
+
+    /* Requesting action from porter(s): override this API if this capability
+       is supported. */
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+ble_error_t Gap::setTxPower(int8_t txPower)
+{
+    /* Avoid compiler warnings about unused variables. */
+    (void)txPower;
+
+    /* Requesting action from porter(s): override this API if this capability
+       is supported. */
+    return BLE_ERROR_NOT_IMPLEMENTED;
+}
+
+void Gap::getPermittedTxPowerValues(
+    const int8_t **valueArrayPP, size_t *countP
+)  {
+    /* Avoid compiler warnings about unused variables. */
+    (void)valueArrayPP;
+    (void)countP;
```

Comment:
Why is this parameter being voided and then used below ?

Reply:
Cut and paste from the old code. I will fix that.

Reply:
Fixed.